### PR TITLE
fix: wait for mids to be indexed

### PR DESCRIPTION
**Pull request by stephhuynh18**
fixes two things:
- we cannot control the order we receive pubsub messages. We cannot expect the most recently indexed mid to be the one we created
- sometimes it takes a while to receive mids created by a remote node. We attempted to use a delay but sometimes it is not enough. So i am using a polling strategy

### CODE REVIEW

File: src/__tests__/indexing.ts [hunk 1]
```diff
@@ -290,17 +291,20 @@ describe('indexing', () => {
...
             .query({ model: TEST_MODEL, last: 100, account: did1.id })
             .then(resultObj => extractDocuments(ceramic2, resultObj))
           expect(did1Results.length).toBeGreaterThanOrEqual(1)
 
-          const lastDid1Doc = did1Results.at(-1) as ModelInstanceDocument
-          expect(lastDid1Doc.id.toString()).toEqual(doc1.id.toString())
-          expect(lastDid1Doc.content).toEqual(doc1.content)
-          expect(StreamUtils.serializeState(lastDid1Doc.state)).toEqual(
+          // We cannot expect that the most recent MID will be the MID created by this test
+          // This is because we may receive and process pubsub messages for docs created by ceramicClient
```
Comment (stbrody) [inline src/__tests__/indexing.ts:302]: what do you mean by "docs created by ceramicClient" here?

File: src/__tests__/indexing.ts [hunk 2]
```diff
@@ -38,6 +39,44 @@ const extractDocuments = (
...
 
+const waitForMidsToBeIndexed = (
+  ceramic: CeramicApi,
+  docs: ModelInstanceDocument[]
+): Promise<Record<string, ModelInstanceDocument>> => {
+  return firstValueFrom(
+    interval(1000).pipe(
+      mergeMap(() => ceramic.index.query({ model: TEST_MODEL, last: 100 })),
+      map(resultObj => extractDocuments(ceramic, resultObj)),
+      map(results =>
```
Comment (stbrody) [inline src/__tests__/indexing.ts:50]: I'm sorry, this broke my brain.
I find this type of complex functional code really hard to follow.  Is there any way to make it easier to follow what this is doing?  Easiest for me would probably be just to write it iteratively instead of functionally, but also maybe comments or breaking it up to smaller helper functions would possibly be another option?

File: src/__tests__/indexing.ts (hunk 1 above)
Comment (stephhuynh18) [reply src/__tests__/indexing.ts:302]: The test that is failing is `Can filter by DID` using `ceramic` (local). The test that runs before this is `Can create and query on same node` using the `ceramicClient` (remote node). Since pubsub messages can be received in a random order, sometimes the `ceramic` instance receives and indexes docs created by the `ceramicClient` from other tests. 

Actually this probably happens with `ceramic` and `ceramicClient` swapped. I'll change the comment to reflect this

File: src/__tests__/indexing.ts (hunk 2 above)
Comment (stephhuynh18) [reply src/__tests__/indexing.ts:50]: yup will do

File: src/__tests__/indexing.ts (hunk 1 above)
Comment (stbrody) [reply src/__tests__/indexing.ts:332]: think you forgot to update this comment?  There's another comment above, on line 206, that looks better now

**Review by stbrody — APPROVED** [review]
lgtm
